src/gprs.c

```diff
@@ -536,9 +538,171 @@ static void pri_context_signal_settings(struct pri_context *ctx,
 				context_settings_append_ipv6);
 }
 
-static void pri_parse_proxy(struct pri_context *ctx, const char *proxy)
+static void set_default_dns(char **dns_servers)
 {
-	char *scheme, *host, *port, *path;
+	struct in_addr server_in_addr;
+	int i;
+
+	/*
+	 * In this function we modify the global variable _res from resolv.h to
+	 * be able to use a specific DNS server to resolve the MMS proxy/MMSC
+	 * address. This is a hack, although used in some places. Also, the IPv6
+	 * case cannot be handled this way. ldns is a possible option
+	 * (http://www.nlnetlabs.nl/projects/ldns/), but it would introduce a
+	 * new dependence. TODO: search alternatives.
+	 */
+	_res.nscount = 0;
```

Comment:
As this isn't super obvious, maybe a note that this is initializing the number of DNS servers?